astroquery/heasarc/core.py

```diff
@@ -1,92 +1,246 @@
 
-    """
-    HEASARC query class.
+    Example Usage:
+    ...
 
-    For a list of available HEASARC mission tables, visit:
-        https://heasarc.gsfc.nasa.gov/cgi-bin/W3Browse/w3catindex.pl
     """
 
-    URL = conf.server
-    TIMEOUT = conf.timeout
-    coord_systems = ['fk5', 'fk4', 'equatorial', 'galactic']
+    # we can move url to Config later
+    VO_URL = conf.VO_URL
+    TAR_URL = conf.TAR_URL
+    S3_BUCKET = conf.S3_BUCKET
+    timeout = conf.timeout
+
+    def __init__(self):
+        """Initialize some basic useful variables"""
+        super().__init__()
+        self._tap = None
+        self._datalink = None
+        self._meta_info = None
+        self._session = None
+
+    @property
+    def tap(self):
+        """TAP service"""
+        if self._tap is None:
+            self._tap = pyvo.dal.TAPService(
+                f'{self.VO_URL}/tap', session=self._session
+            )
+            self._session = self._tap._session
+        return self._tap
+
+    @property
+    def _meta(self):
+        """Queries and holds meta-information about the tables.
+
+        This is a table that holds useful information such as
+        the list of default columns per table, the reasonable default
+        search radius per table that is appropriate for a mission etc.
+        Instead of making a server call for each of that type information,
+        we do a single one and then post-process the resulting table.
+
+        These are not meant to be used directly by the user.
+        """
+        if self._meta_info is None:
+            query = (
+                "SELECT split_part(name, '.', 1) AS table, "
+                "split_part(name, '.', 2) AS par, "
+                "CAST(value AS DECIMAL) AS value "
+                "FROM metainfo "
+                "WHERE (type = 'parameter' and relation = 'order') "
+                "OR relation LIKE 'defaultSearchRadius' "
+                "ORDER BY value"
+            )
+            self._meta_info = self.query_tap(query).to_table()
+            self._meta_info['value'] = np.array(
+                self._meta_info['value'], np.float32)
+            self._meta_info = self._meta_info[self._meta_info['value'] > 0]
+        return self._meta_info
+
+    def _get_default_cols(self, table_name):
+        """Get a list of default columns for a table
+
+        Parameters
+        ----------
+        table_name: str
+            The name of table as a str
+
+        Return
+        ------
+        a list of column names
 
-    def query_async(self, request_payload, *, cache=True, url=None):
         """
-        Submit a query based on a given request_payload. This allows detailed
-        control of the query to be submitted.
+        meta = self._meta[
+            (self._meta['table'] == table_name)
+            & (self._meta['par'] != '')
+        ]
+        meta.sort('value')
+        defaults = meta['par']
+        return defaults
+
+    def get_default_radius(self, table_name):
```

Comment:
make this private, too, similar to `_get_default_cols`

Reply:
Done